streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -70,9 +70,8 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
     runningBatchInfos(batchStarted.batchInfo.batchTime) = batchStarted.batchInfo
     waitingBatchInfos.remove(batchStarted.batchInfo.batchTime)
 
-    batchStarted.batchInfo.receivedBlockInfo.foreach { case (_, infos) =>
-      totalReceivedRecords += infos.map(_.numRecords).sum
-    }
+    // TODO. this should be fixed when input stream is not receiver based stream.
+    totalReceivedRecords += batchStarted.batchInfo.streamIdToNumRecords.values.sum
```

Comment:
Yes we need to. Receiver or not, all the data has been received by the
system.
On Apr 30, 2015 1:10 AM, "Saisai Shao" <notifications@github.com> wrote:

> In
> streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala
> <https://github.com/apache/spark/pull/5680#discussion_r29410293>:
>
> > @@ -70,9 +70,8 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
> >      runningBatchInfos(batchStarted.batchInfo.batchTime) = batchStarted.batchInfo
> >      waitingBatchInfos.remove(batchStarted.batchInfo.batchTime)
> >
> > -    batchStarted.batchInfo.receivedBlockInfo.foreach { case (_, infos) =>
> > -      totalReceivedRecords += infos.map(_.numRecords).sum
> > -    }
> > +    // TODO. this should be fixed when input stream is not receiver based stream.
> > +    totalReceivedRecords += batchStarted.batchInfo.streamIdToNumRecords.values.sum
>
> Yes, will do. Also have one concern, if the batchStarted is not a
> receiver-based batchInfo, so do we need to count this records into
> totalReceivedRecords when batch is just started.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/5680/files#r29410293>.
>